### PR TITLE
Update PWA plugin API calls for v0.6

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

This prevents deprecation warnings from being raised when using PWA v0.6.

See https://github.com/GoogleChromeLabs/pwa-wp/pull/338

## Checklist

- [ ] My pull request is addressing an open issue (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/wiki/Engineering-Guidelines) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by westonruter** [comment]
Actually, the old method signature of `WP_Service_Worker_Caching_Routes::register( $route, $args )` does not cause a deprecation notice. Instead, it is transparently converted to the new signature: `WP_Service_Worker_Caching_Routes::register( $route, $strategy, $args )`:

https://github.com/GoogleChromeLabs/pwa-wp/blob/17e23ef28e9224c5cdbc839596377e10c7b158a2/wp-includes/components/class-wp-service-worker-caching-routes.php#L103-L108

So no need for this to be included in v2.0.8.

**Comment by westonruter** [comment]
When on the `develop` branch, there are deprecation warnings:

```
$ phpunit tests/php/test-class-amp-service-worker.php
Installing...
Running as single site... To run multisite, use -c tests/phpunit/multisite.xml
Not running ajax tests. To execute these, use --group ajax.
Not running ms-files tests. To execute these, use --group ms-files.
Not running external-http tests. To execute these, use --group external-http.
PHPUnit 5.7.27 by Sebastian Bergmann and contributors.

....FF....                                                        10 / 10 (100%)

Time: 4.41 seconds, Memory: 38.50MB

There were 2 failures:

1) Test_AMP_Service_Worker::test_add_image_caching
Unexpected incorrect usage notice for WP_Service_Worker_Caching_Routes::register
Failed asserting that an array is empty.

/app/public/core-dev/tests/phpunit/includes/abstract-testcase.php:516
/app/public/core-dev/tests/phpunit/includes/abstract-testcase.php:528

2) Test_AMP_Service_Worker::test_add_google_fonts_caching
Unexpected incorrect usage notice for WP_Service_Worker_Caching_Routes::register
Failed asserting that an array is empty.

/app/public/core-dev/tests/phpunit/includes/abstract-testcase.php:516
/app/public/core-dev/tests/phpunit/includes/abstract-testcase.php:528

FAILURES!
Tests: 10, Assertions: 32, Failures: 2.
```

These are absent when testing on this branch.

**Comment by westonruter** [comment]
We will need this to go along with PWA 0.6.0 since there _will_ be incorrect usage notices otherwise. Specifically, there will the deprecation notice:

> The plugins configuration key is obsolete. Define Workbox plugin configuration at the top level.

See [logic](https://github.com/GoogleChromeLabs/pwa-wp/blob/6c3500fe8893a2e9e5509113a712f7648bf43d02/wp-includes/components/class-wp-service-worker-caching-routes.php#L190-L198) from the PWA plugin's `\WP_Service_Worker_Caching_Routes::normalize_configuration()`. 

Without this PR, the `add_image_caching` and `add_google_fonts_caching` methods use that deprecated `plugins` key. For more on why the `plugins` key is deprecated, see https://github.com/GoogleChromeLabs/pwa-wp/pull/338:

> There is also this `plugins` array key which is essentially how Workbox.js is passed the configuration information to these [Workbox plugins](https://developers.google.com/web/tools/workbox/modules/workbox-strategies#using_plugins), but in the WordPress context this is confusing and unnecessary. So the array is flattened so that the plugin names can be supplied as top-level keys, so `$args['expiration']` instead of `$args['plugins']['expiration']`.

**Review by pierlon — APPROVED** [review]
Tested the changes with the PWA plugin v0.6.0-RC1 and noticed no deprecation notices or PHP warnings :+1:.

File: includes/class-amp-service-worker.php [hunk 1]
```diff
@@ -65,6 +65,27 @@ public static function init() {
...
+	 * @param string                    $strategy        Strategy name.
+	 * @param array                     $args            Strategy args.
+	 * @param array                     $plugins         Plugins.
+	 */
+	private static function register_caching_route( WP_Service_Worker_Scripts $service_workers, $route, $strategy, $args = [], $plugins = [] ) {
+		$caching_routes = $service_workers->caching_routes();
+		if ( defined( 'PWA_VERSION' ) && version_compare( PWA_VERSION, '0.6', '<' ) ) {
+			$args['strategy'] = $strategy;
+			$args['plugins']  = $plugins;
+			$caching_routes->register( $route, $args );
```
Comment (pierlon) [inline includes/class-amp-service-worker.php:82]: A really minor nitpick: This `register` method signature no longer matches the updated stub. I wonder if there is a way to indicate the previous method signatures on the stub.

Comment (westonruter) [reply includes/class-amp-service-worker.php:82]: The new stub is:

```
register( $route, $strategy, $args = array() )
```

So doesn't it match? The `$args` has a default value of `array()` so it matches, right?

Comment (westonruter) [reply includes/class-amp-service-worker.php:82]: image

image

The description for the `$strategy` actually indicates the deprecated usage, and it still allows `array`.

Comment (pierlon) [reply includes/class-amp-service-worker.php:82]: I see that `$strategy` actually indicates the deprecated usage, so no worries here.

**Comment by westonruter** [comment]
I've deployed PWA 0.6.0-RC1 and AMP 2.0.9-RC1 to https://amp-wp.org/ and the AMP assets are being cached in the service worker as expected.
